### PR TITLE
CIV-10448 S2S refactor

### PR DESCRIPTION
**Pull request by vasudevganesanhmcts**
initial commit with the relevant changes

    - introduced authorisation service
    - changes on payment call back service to check and throw relevant exception

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Screenshot attached for the Evidence of manual testing
- [ ] Functional, Smoke and API tests have been run locally
- [ ] Full gradle build and run tests with coverage have been completed (if needed)

Please remove this line and everything above and fill the following sections:

### Testing branch and instruction ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/service/AuthorisationService.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    private final IdamClient idamClient;
+
+    private UserInfo userInfo;
+
+    public Boolean authoriseService(String serviceAuthHeader) {
+        String callingService;
+        try {
+            log.info("service auth token header... {}", serviceAuthHeader);
+            String bearerJwt = serviceAuthHeader.startsWith("Bearer ") ? serviceAuthHeader : "Bearer " + serviceAuthHeader;
+            log.info("bearer token header... {}", bearerJwt);
```
Comment (sampankumar) [inline src/main/java/uk/gov/hmcts/reform/civil/service/AuthorisationService.java:33]: May be not log it.

File: src/main/java/uk/gov/hmcts/reform/civil/service/AuthorisationService.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            //do nothing
+            log.error("S2S token is not authorised" + ex);
+        }
+        return false;
+    }
+
+    public Boolean authoriseUser(String authorisation) {
+        try {
+            userInfo = idamClient.getUserInfo(authorisation);
+            log.info("Userinfo {}", userInfo);
```
Comment (sampankumar) [inline src/main/java/uk/gov/hmcts/reform/civil/service/AuthorisationService.java:51]: Same here

File: src/main/java/uk/gov/hmcts/reform/civil/service/AuthorisationService.java (hunk 1 above)
Comment (vasudevganesanhmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/service/AuthorisationService.java]: done

File: src/main/java/uk/gov/hmcts/reform/civil/service/AuthorisationService.java (hunk 2 above)
Comment (vasudevganesanhmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/service/AuthorisationService.java]: done

**Review by sampankumar — DISMISSED** [review]

**Review by deepthidoppalapudihmcts — DISMISSED** [review]

**Review by deepthidoppalapudihmcts — APPROVED** [review]

**Review by sampankumar — APPROVED** [review]

**Pull request merged**